gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/MetricSchemaMarkdownPlugin.java

```diff
@@ -32,33 +34,41 @@ public void apply(Project project) {
         project.getPluginManager().apply(MetricSchemaPlugin.class);
 
         TaskProvider<CreateMetricsManifestTask> createMetricsManifest =
-                project.getTasks().named(MetricSchemaPlugin.CREATE_METRICS_MANIFEST, CreateMetricsManifestTask.class);
+                project.getTasks().named(CreateMetricsManifestTask.NAME, CreateMetricsManifestTask.class);
 
-        TaskProvider<CheckMetricMarkdownTask> checkMetricsMarkdown = project.getTasks()
-                .register(MetricSchemaPlugin.CHECK_METRICS_MARKDOWN, CheckMetricMarkdownTask.class, task -> {
-                    task.getManifestFile().set(createMetricsManifest.flatMap(CreateMetricsManifestTask::getOutputFile));
-                    task.setMarkdownFile(project.file("metrics.md"));
-                    task.dependsOn(createMetricsManifest);
+        Provider<RegularFile> manifestFile = createMetricsManifest.flatMap(CreateMetricsManifestTask::getOutputFile);
+        Provider<String> localCoordinates = project.provider(() -> "" + project.getGroup() + ':' + project.getName());
+        RegularFile markdownFile = project.getLayout().getProjectDirectory().file("metrics.md");
+
+        TaskProvider<GenerateMetricMarkdownTask> generateMetricMarkdownTask = project.getTasks()
+                .register(GenerateMetricMarkdownTask.NAME, GenerateMetricMarkdownTask.class, task -> {
+                    task.setGroup(MetricSchemaPlugin.TASK_GROUP);
+                    task.getManifestFile().set(manifestFile);
+                    task.getLocalCoordinates().set(localCoordinates);
+                    task.getMarkdownFile().set(markdownFile);
                 });
 
-        project.getTasks()
-                .register(MetricSchemaPlugin.GENERATE_METRICS_MARKDOWN, GenerateMetricMarkdownTask.class, task -> {
-                    task.getManifestFile().set(createMetricsManifest.flatMap(CreateMetricsManifestTask::getOutputFile));
-                    task.getOutputFile().set(project.file("metrics.md"));
-                    task.dependsOn(createMetricsManifest);
-                    // Avoid a race when both checkMetricsMarkdown and generateMetricsMarkdown are requested
-                    task.mustRunAfter(checkMetricsMarkdown);
```

Comment:
I've inverted this `mustRunAfter` ordering. If you're running `generate` and `check` you almost certainly want `generate` to run first.